<?xml version="1.0" encoding="UTF-8"?>

<launch>

  <arg name="config_yaml" default="eagleye_config.yaml"/>
  <arg name="topic_yaml" default="eagleye_topic.yaml"/>
 
  <include file="$(find-pkg-share nmea2fix)/launch/nmea2fix.launch"/>
 

  <node pkg="eagleye_rt" name="velocity_scale_factor_node" exec="velocity_scale_factor">
        <param from="$(find-pkg-share eagleye_rt)/config/$(var config_yaml)"/>
        <param from="$(find-pkg-share eagleye_rt)/config/$(var topic_yaml)"/>
  </node>
  <node pkg="eagleye_rt" name="yawrate_offset_stop_node" exec="yawrate_offset_stop">
        <param from="$(find-pkg-share eagleye_rt)/config/$(var config_yaml)"/>
        <param from="$(find-pkg-share eagleye_rt)/config/$(var topic_yaml)"/>
  </node>
  <node pkg="eagleye_rt" name="yawrate_offset_node_1st" exec="yawrate_offset" args="1st">
        <param from="$(find-pkg-share eagleye_rt)/config/$(var config_yaml)"/>
        <param from="$(find-pkg-share eagleye_rt)/config/$(var topic_yaml)"/>
  </node>
  <node pkg="eagleye_rt" name="yawrate_offset_node_2nd" exec="yawrate_offset" args="2nd">
        <param from="$(find-pkg-share eagleye_rt)/config/$(var config_yaml)"/>
        <param from="$(find-pkg-share eagleye_rt)/config/$(var topic_yaml)"/>
  </node>
  <node pkg="eagleye_rt" name="heading_node_1st" exec="heading" args="1st">
        <param from="$(find-pkg-share eagleye_rt)/config/$(var config_yaml)"/>
        <param from="$(find-pkg-share eagleye_rt)/config/$(var topic_yaml)"/>
  </node>
  <node pkg="eagleye_rt" name="heading_node_2nd" exec="heading" args="2nd">
        <param from="$(find-pkg-share eagleye_rt)/config/$(var config_yaml)"/>
        <param from="$(find-pkg-share eagleye_rt)/config/$(var topic_yaml)"/>
  </node>
  <node pkg="eagleye_rt" name="heading_node_3rd" exec="heading" args="3rd">
        <param from="$(find-pkg-share eagleye_rt)/config/$(var config_yaml)"/>
        <param from="$(find-pkg-share eagleye_rt)/config/$(var topic_yaml)"/>
  </node>
  <node pkg="eagleye_rt" name="heading_interpolate_node_1st" exec="heading_interpolate" args="1st">
        <param from="$(find-pkg-share eagleye_rt)/config/$(var config_yaml)"/>
        <param from="$(find-pkg-share eagleye_rt)/config/$(var topic_yaml)"/>
  </node>
  <node pkg="eagleye_rt" name="heading_interpolate_node_2nd" exec="heading_interpolate" args="2nd">
        <param from="$(find-pkg-share eagleye_rt)/config/$(var config_yaml)"/>
        <param from="$(find-pkg-share eagleye_rt)/config/$(var topic_yaml)"/>
  </node>
  <node pkg="eagleye_rt" name="heading_interpolate_node_3rd" exec="heading_interpolate" args="3rd">
        <param from="$(find-pkg-share eagleye_rt)/config/$(var config_yaml)"/>
        <param from="$(find-pkg-share eagleye_rt)/config/$(var topic_yaml)"/>
  </node>
  <node pkg="eagleye_rt" name="slip_angle_node" exec="slip_angle">
        <param from="$(find-pkg-share eagleye_rt)/config/$(var config_yaml)"/>
        <param from="$(find-pkg-share eagleye_rt)/config/$(var topic_yaml)"/>
  </node>
  <node pkg="eagleye_rt" name="distance_node" exec="distance">
        <param from="$(find-pkg-share eagleye_rt)/config/$(var config_yaml)"/>
        <param from="$(find-pkg-share eagleye_rt)/config/$(var topic_yaml)"/>
  </node>
  <node pkg="eagleye_rt" name="trajectory_node" exec="trajectory">
        <param from="$(find-pkg-share eagleye_rt)/config/$(var config_yaml)"/>
        <param from="$(find-pkg-share eagleye_rt)/config/$(var topic_yaml)"/>
  </node>
  <node pkg="eagleye_rt" name="position_node" exec="position">
        <param from="$(find-pkg-share eagleye_rt)/config/$(var config_yaml)"/>
        <param from="$(find-pkg-share eagleye_rt)/config/$(var topic_yaml)"/>
  </node>
  <node pkg="eagleye_rt" name="position_interpolate_node" exec="position_interpolate">
        <param from="$(find-pkg-share eagleye_rt)/config/$(var config_yaml)"/>
        <param from="$(find-pkg-share eagleye_rt)/config/$(var topic_yaml)"/>
  </node>
  <node pkg="eagleye_rt" name="smoothing_node" exec="smoothing">
        <param from="$(find-pkg-share eagleye_rt)/config/$(var config_yaml)"/>
        <param from="$(find-pkg-share eagleye_rt)/config/$(var topic_yaml)"/>
  </node>
  <node pkg="eagleye_rt" name="height_node" exec="height">
        <param from="$(find-pkg-share eagleye_rt)/config/$(var config_yaml)"/>
        <param from="$(find-pkg-share eagleye_rt)/config/$(var topic_yaml)"/>
  </node>
  <node pkg="eagleye_rt" name="angular_velocity_offset_stop_node" exec="angular_velocity_offset_stop">
        <param from="$(find-pkg-share eagleye_rt)/config/$(var config_yaml)"/>
        <param from="$(find-pkg-share eagleye_rt)/config/$(var topic_yaml)"/>
  </node>
  <node pkg="eagleye_rt" name="correction_imu" exec="correction_imu">
        <param from="$(find-pkg-share eagleye_rt)/config/$(var config_yaml)"/>
        <param from="$(find-pkg-share eagleye_rt)/config/$(var topic_yaml)"/>
  </node>
  <node pkg="eagleye_rt" name="monitor" exec="monitor" output="screen">
        <param from="$(find-pkg-share eagleye_rt)/config/$(var config_yaml)"/>
        <param from="$(find-pkg-share eagleye_rt)/config/$(var topic_yaml)"/>
  </node>

</launch>
